docs/spec/spec.md

```diff
@@ -0,0 +1,330 @@
+- Subscriber: Discovers the `hub` and `topic` URL given a resource URL, subscribes to updates at the `hub`, and accepts
+  content distribution requests from the `hub`.
+- Hub: Handles subscription requests and distributes the content to subscribers when the corresponding topic URL has
+  been updated.
+
+`WebSub` is a library which is derived from the WebSub specification which could be used by developers to implement
+WebSub compliant `subscriber` services.  
+
+## 2. Subscriber
+
+WebSub `subscriber` will subscribe to a `hub` to receive content updates for a `topic`.  
+
+It has the following capabilities:
+* Discover the `hub` and the `topic` given a resource URL
+* Subscribe to content updates for a `topic` in a `hub`
+* Accept content distribution requests from the `hub`
+
+The `subscriber` is designed in the form of `listener` and `Subscriber Service`.
+- `websub:Listener`: A listener end-point to which `websub:SubscriberService` could be attached.
+- `websub:SubscriberService`: An API service, which receives WebSub events.
+
+### 2.1. Listener
+
+The `websub:Listener` opens the given port and attaches the provided `websub:SubscriberService` object to the given
+service-path. `websub:Listener` can be initialized either by providing a port with listener configurations or by
+providing an `http:Listener`.
+
+#### 2.1.1. Configuration
+
+When initializing a `websub:Listener`, following configurations could be provided.
+```ballerina
+# Provides a set of configurations for configure the underlying HTTP listener of the WebSub listener.
+# 
+# + gracefulShutdownPeriod - The time period in seconds to wait for unsubscription verification
+public type ListenerConfiguration record {|
+    *http:ListenerConfiguration;
+    decimal gracefulShutdownPeriod = 20;
+|};
+```
+
+For more details on the available configurations please refer [`http:ListenerConfiguration`](https://lib.ballerina.io/ballerina/http/latest/records/ListenerConfiguration).
+
+#### 2.1.2. Initialization
+
+The `websub:Listener` could be initialized by providing either a port with `websub:ListenerConfiguration` or by
+providing an `http:Listener`.
+```ballerina
+# Initiliazes `websub:Listener` instance.
+# ```ballerina
+# listener websub:Listener websubListenerEp = check new (9090);
+# ```
+#
+# + listenTo - Port number or a `http:Listener` instance
+# + config - Custom `websub:ListenerConfiguration` to be provided to underlying HTTP Listener
+# + return - The `websub:Listener` or an `websub:Error` if the initialization failed
+public isolated function init(int|http:Listener listenTo, *ListenerConfiguration config) returns websub:Error? 
+```
+
+#### 2.1.3. Methods
+
+Following APIs should be available in the `websub:Listener` to dynamically attach `websub:SubscriberService` objects to 
+it.
+```ballerina
+# Attaches the provided `websub:SubscriberService` to the `websub:Listener`.
+# ```ballerina
+# check websubListenerEp.attach('service, "/subscriber");
+# ```
```

Comment:
```suggestion
```

Reply:
Let's remove these inner quotes. Due to the out quotes, inner quotes will not function I think. Check the other places as well